### ISSUE TITLE
Right-click / View View / No Database Selected if view references another view

### ISSUE THREAD

**Issue by GeneralProtectionFault** [dbeaver/dbeaver#23210]
### Description

This is an issue that I have seen over the last couple years I have used DBeaver.  I do not know if it is exclusive thus, but I most notice it connecting to a MySQL 5.7 database (happens to be an RDS in AWS).

Side note:  Even if it's successful in showing the view (some views work, again, I think it's views that do not reference another view), this dropdown still incorrectly says <N/A>, and still shows empty.

### DBeaver Version

Community Edition 24.0.0.202403091004

### Operating System

WIndows 10

### Database and driver

MySQL (5.7)
com.mysql.cj.jdbc.CommentClientInfoProvider
com.mysql.cj.jdbc.ha.StandardLoadBalanceExceptionChecker

### Steps to reproduce

- Connect to the (MySQL) database
- Expand the views in the Database Navigator
- Right-click on a view *that references another view*, click "View View"
- SQL Error [1046] [3D000]: No database selected
- Clicking the database selection dropdown (< N/A > ), it shows as Empty, even though there are databases available

### Additional context

Connecting w/ SSL is required WITHOUT certificate verification (protocol only)

**Comment by Matvey16** [comment]
@GeneralProtectionFault Hello. Thank you for the report. Can you specify what kind of an issue does this cause. YOu can't view data when you open this View? Or is the problem with < N /A > info in toolbar? Dose it happen consistently on the same view, or sometimes?

**Comment by GeneralProtectionFault** [comment]
Sure.  The main problem is on some views, right clicking and selecting "View View" results in this error:
ERROR WHILE READING SOURCE:
SQL Error [1046] [3D000]: No database selected

...so I cannot view the data or the source.  The source being the serious problem, because then I need to load up *cringe* Workbench just to see the view's source.

So, going by that, I try to select the database manually from the dropdown (where it says < N/A >), and the dropdown does not have any options to select, even though there are a few databases to choose from in reality.

So it does happen consistently on the same views, and these views are copied on dev & production servers, and the behavior is consistent across the servers.  However, I'm now not sure about my guess that it's related to the view referencing another view.  After some digging, I found that some views which referenced other views did, in fact, work.  Might be joining to another view within a sub-query in a certain way, I'm not sure, but it's definitely consistent that some combination of factors in a view cause the behavior.  It involves some client data, so I can't just plop everything in here, but I'm happy to try any specific experiments or provide any additional info that I can.

The secondary issue is not so critical, but still what I would call an "error" and maybe it would be related in the code, just guessing.  So, on the views that DO work (or tables for that matter), if you right click, then "view" the source or data, the dropdown that shows the selected database still shows < N/A >.  This isn't accurate, of course, since the database is whatever one the table or view you right-clicked on to see the definition of.  It would make sense not to be able to select it since you're looking at an object specific to a database, but N/A doesn't make sense, and just mentioning it in case there's any similar code "looking" for the database that would elucidate the problem.

**Comment by Matvey16** [comment]
@GeneralProtectionFault Thank you for your answer. I tried reproducing it with basic select of another view in another database with no luck. Can you give me a source of any of those problematic views with every name or data omitted? I just want to see sql in it, you can change any name in it with db1/db2/view1/table2 ans if it has data in filtering just replace it with 'data1'

Regarding database selector. Ii is expected to see n/a in any scenario other than when focus is on SQL editor, because it is actualy used only there. It doesn't affect data or metadata editor in any way

**Comment by GeneralProtectionFault** [comment]
Yeah I expect it won't show up with any basic select.  Ok, gimme a little time, I will see if I can pin down anything, and if not, I'll at least get you the schema and view/sql structure.

**Comment by GeneralProtectionFault** [comment]
Ok I think this will help more.  So I did actually find some inconsistent behavior regarding viewing the **data** (the source still consistently borks).  So, DBeaver remembers what I had open, so I opened the source (failed) on the offending view, then closed DBeaver, then opened it again with the trace on the command line (dbeaver-cli -Ddbeaver.trace.enabled=true).    The argument did nothing, apparently, with this in the output: 
Error parsing command line: Unrecognized option: -Ddbeaver.trace.enabled=true 
So I don't understand the docs in that case, but we still have the command line output.

This starts it up and errors immediately as it tries to restore the open view I had up when I had closed it (only one of course).

I'm attaching 4 files.
[DBeaver_Startup_ViewSourceError.txt](https://github.com/dbeaver/dbeaver/files/14841485/DBeaver_Startup_ViewSourceError.txt)

[DBeaver_ViewBug1.sql.txt](https://github.com/dbeaver/dbeaver/files/14841301/DBeaver_ViewBug1.sql.txt)
[DBeaver_ViewBugReferencedView1.sql.txt](https://github.com/dbeaver/dbeaver/files/14841302/DBeaver_ViewBugReferencedView1.sql.txt)
[DBeaver_ViewBugReferencedView2.sql.txt](https://github.com/dbeaver/dbeaver/files/14841303/DBeaver_ViewBugReferencedView2.sql.txt)

The ViewSourceError is the console output from me starting up DBeaver and getting the error as I described.
The ViewBug1 file is the SQL for the offending view.  I search-and-replaced to obfuscate the naming as you mentioned.
In this one, I named the columns after their data types just to give an idea in case it helps.
You'll see it reference vw_One & vw_Two, the sql of which is similarly in the other two files.

Sorry about all the parentheses, but thanks MySQL for not storing the text and decompiling the view :-P.
* One thing that I'm starting to be suspicious of...  is the nested subquery, or maybe joins/nested joins on subqueries.

**Comment by MrYutz** [comment]
I can confirm this is happening to me as well.  On my Mysql `5.7.33-0ubuntu0.16.04.1-log` I get the error.

On the replication slave, which is running `8.0.28` does not produce the error. 

Obviously the problem is that I need to update the view on the master, and I can't via dbeaver due to the error "SQL Error [1046] [3D000]: No database selected"

I have set the default database, selected data from the view via a sql prompt, but have not been able to edit the view or alter the error on mysql 5.7.

I can work around the issue by getting the view structure from the slave, and then updating the master via a sql query.

**Comment by GeneralProtectionFault** [comment]
I think I have a possible clue here...  I actually managed to "sort of" get this error in MySQL Workbench itself.  It still appears to only affect certain views, so I still think it is necessary to have that nested nonsense or whatever is tripping it up.

(In Workbench):
1.  Create a new connection to the database in question.  Do NOT select a default schema (this is the crux, I think).
2.  Navigate to the offending view, without double-clicking/bolding the database you're in, just expand it w/ the little triangle to the left of the database, then Views.
3. Click the little wrench icon to go to the DDL on the offending view, and error:
![image](https://github.com/dbeaver/dbeaver/assets/29645865/cda51e59-5ee1-4b3a-96e9-fa7825841d59)
If you try clicking the table icon (to select from it), you get this:
Error Code: 1046. No database selected
Select the default DB to be used by double-clicking its name in the SCHEMAS list in the sidebar.

Once you double click a database on the left (it appears bolded), this error goes away for good.  Since Workbench remembers your settings, the next time you log in, the database you last selected is selected, which is why this only shows up when a new connection is made.

I'm having trouble replicating this part (maybe something cached somewhere):
....but the first time, I did this, I got an additional error pop-up, that seemed to be complaining of permissions--that the account I had did not have permissions to select from what seemed to be system tables...
Another footnote here is that AWS RDS is not necessarily exactly in parity with running MySQL server on bare metal, in case this is relevant:
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Concepts.FeatureSupport.html#MySQL.Concepts.Features

So, I'm thinking the problem might be...  Essentially, without that database selected, workbench tries to *list* the databases using the credentials of the logged-in user, which does not directly have permission to get that from all databases.  This is a guess, of course--but it's trying to do something it wouldn't try to do if a database were selected.

Extra info:
If I click the little 'i' information icon, that works and shows view stats, but the DDL tab is blank.

**Comment by GeneralProtectionFault** [comment]
Hi @E1izabeth.  I noticed this was assigned, but the wait for review tag is still on it.  Should that be removed?  I figure that might be the root of the crickets :).

**Comment by E1izabeth** [comment]
Hi, @GeneralProtectionFault
It's assigned for review to reproduce the issue.
